xword.py

```diff
@@ -1,21 +1,45 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
-"""Crossword Solver Program"""
+"""
+Crossword Solver Program
 
-__author__ = "???"
+Description: displays possible words to fit crossword puzzle location based on
+dictionary of words
 
-# YOUR HELPER FUNCTION GOES HERE
+Author: Aaron Jackson
+Github: TimeApollo
+"""
+
+__author__ = "TimeApollo"
+import re
+
+
+def word_checker(in_word, words):
+    """checks in_word against words list, to see if any match pattern"""
+    """returns list as well as printing all matches"""
```

Comment:
For future reference, this docstring hints that the function is doing two things, straying away from the 'separation of concerns' principle

Reply:
I agree with that. i was confused on what was wanted and then came to the decision to print and return from the same function. should have split the printing out on its own and just returned the list.